### PR TITLE
Better fix for iteration issue

### PR DESCRIPTION
**Pull request by JB1984**
fix iteration issue in ising_hamiltonian function

**Related GitHub Issues:** See #45 for a discussion on this bug.

### CODE REVIEW

**Comment by co9olguy** [comment]
Thanks @JB1984! We'll wait for @mariaschuld to give a check before merging, since she wrote this demo.

File: .circleci/config.yml [hunk 1]
```diff
@@ -65,7 +65,13 @@ jobs:
...
       - run:
           name: Comment artifacts
-          command: pip3 install pygithub --user && python3 .circleci/comment.py
+          command: |
+            if [[ -z "${GH_AUTH_TOKEN}" ]]; then
+              echo "GitHub token not available, skipping uploading artifacts..."
+            else
+              echo "Commenting artifacts..."
+              pip3 install pygithub --user && python3 .circleci/comment.py
+            fi
```
Comment (mariaschuld) [inline .circleci/config.yml]: @josh146 I had an unintentional change like this happen before - do you know what causes it? Is it just the circleci test that got updated somewhere?

In other words, safe to merge?

Comment (josh146) [reply .circleci/config.yml]: Sorry @mariaschuld for the confusion, this change was made by me in commits 
42e858b and 833ea05! Initially, CircleCI was failing --- I realised this was a bug in my configuration due to not taking into account PRs from forks.

Comment (josh146) [reply .circleci/config.yml]: Safe to merge :slightly_smiling_face:

**Pull request merged**